Package builds find with our version of setuptools_scm, so use it.

--- a/setup.py
+++ b/setup.py
@@ -115,7 +115,7 @@
         'doc': read_requirements('doc'),
         'license': read_requirements('license'),
     },
-    setup_requires=['setuptools_scm<6.0'],
+    setup_requires=['setuptools_scm'],
     use_scm_version=True,
 
     # If there are data files included in your packages that need to be
